File: Examen1/2026_1/Plantillas/PlantillaC#/TestSoft/TestSoftPersistance/DAO/IDAO.cs
namespace TestSoftPersistance.DAO
{
    public interface IDAO<T>
    {
        int Insertar(T objeto);
        int Modificar(T objeto);
        int Eliminar(int id);
        T BuscarPorId(int id);
        List<T> ListarTodos();
    }
}

File: Lab11y14/Laboratorio14_20223796/Backend/fifasoft_solution/fifasoft_persistence/src/pe/edu/pucp/fifasoft/persistance/dao/IDAO.java
package pe.edu.pucp.fifasoft.persistance.dao;

import java.util.List;

// Interfaz genérica CRUD. id como int (el dominio Alumno/Examen usa PK entera autoincrement).
public interface IDAO<T> {
    int insertar(T objeto);
    int modificar(T objeto);
    int eliminar(int id);
    T buscarPorId(int id);
    List<T> listarTodos();
}
